path_role.go

```diff
@@ -302,6 +309,17 @@ func (b *kubeAuthBackend) pathRoleCreateUpdate(ctx context.Context, req *logical
 		role.Audience = audience.(string)
 	}
 
+	if source, ok := data.GetOk("alias_name_source"); ok {
+		if err := validateAliasNameSource(source.(string)); err != nil {
+			return logical.ErrorResponse(err.Error()), nil
+		}
+		role.AliasNameSource = source.(string)
+	} else if role.AliasNameSource == aliasNameSourceUnset {
+		if s, ok := data.Schema["alias_name_source"]; ok {
```

Comment:
I don't think you have to reach into `data.Schema` or fish out the default value through `s.Default` directly. `data.Get` will attempt to get the default value if it's unset or absent so we should be using that instead.

Reply:
sweet, thanks!

Reply:
fixed in https://github.com/hashicorp/vault-plugin-auth-kubernetes/pull/110/commits/40ecafed3570dcfd269833c7834264757cbed356